configuration/secrets.rst

```diff
@@ -0,0 +1,378 @@
+as credentials, passwords, tokens, API keys without exposing them.
+
+.. note::
+
+    In order to use Symfony's secrets, you will need the `libsodium`_ PHP
+    extension or use the `sodium_compat`_ package.
+
+.. _secrets-configuration:
+
+Configuration
+-------------
+
+By Default, secrets are enabled by the Framework. The default behaviors can be
+configured:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/framework.yaml
+        framework:
+            secrets:
+                #vault_directory: '%kernel.project_dir%/config/secrets/%kernel.environment%'
+                #local_dotenv_file: '%kernel.project_dir%/.env.local'
+                #decryption_env_var: 'base64:default::SYMFONY_DECRYPTION_SECRET'
+
+    .. code-block:: xml
+
+            <!-- config/packages/framework.xml -->
+            <?xml version="1.0" encoding="UTF-8" ?>
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:framework="http://symfony.com/schema/dic/framework"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services https://symfony.com/schema/dic/services/services-1.0.xsd
+                    http://symfony.com/schema/dic/framework https://symfony.com/schema/dic/framework/framework-1.0.xsd"
+            >
+                <framework:config secret="%env(APP_SECRET)%">
+                    <framework:secrets
+                        vault_directory="%kernel.project_dir%/config/secrets/%kernel.environment%"
+                        local_dotenv_file="%kernel.project_dir%/.env.local"
+                        decryption_env_var="base64:default::SYMFONY_DECRYPTION_SECRET"
+                    />
+                </framework:config>
+            </container>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
+        $container->loadFromExtension('framework', [
+            'secrets' => [
+                // 'vault_directory' => '%kernel.project_dir%/config/secrets/%kernel.environment%',
+                // 'local_dotenv_file' => '%kernel.project_dir%/.env.local',
+                // 'decryption_env_var' => 'base64:default::SYMFONY_DECRYPTION_SECRET',
+            ],
+        ]);
+
+.. _secrets-generate-keys:
+
+Generate Encryption Keys
+------------------------
+
+In order to encrypt and decrypt **secrets**, symfony needs **encryption keys**.
+This can be done with the provided command ``secrets:generate-keys``.
+
+.. code-block:: terminal
+
+    $ APP_ENV=prod php bin/console secrets:generate-keys
+
+This command generates a new pair of asymetric **encryption keys** in
+``%kernel.project_dir%/config/secrets/%kernel.environment%``.
+The **encryption key** stored in ``%kernel.environment%.sodium.encrypt.key``
+file is used to encrypt secrets when developer adds or updates secrets. This
+key can be commited and publicly shared. The **decryption key** stored in
```

Comment:
committed
I would remove "and publicly shared" - that doesn't bring more clarity to me
is stored in the ... file.